task2/interface_manager.py

```diff
@@ -0,0 +1,95 @@
+import json
+import os
+from collections import defaultdict
+from typing import List, Dict
+
+import requests
+from marshmallow import EXCLUDE
+
+from argument_parser import arg_pars
+from data_type import Interface
+from utils import get_logger
+
+
+def protocol(ssl):
+    if ssl:
+        return 'https'
+    else:
+        return 'http'
+
+
+def url_check(url):
+    try:
+        r = requests.get(url)
+        r.raise_for_status()
+    except Exception as e:
+        raise SystemExit(e)
+
+
+class InterfaceManager:
+    def __init__(self, api_url, logger):
+        self.api_url = api_url
+        self.interface_schema = Interface.schema()
+        self.parsed_data = self._get_all_interfaces()
```

Comment:
@patrikprovaznik please put this to separate method which makes it available to call it at any point.. I know I told you to have it in here only but at least app will make little more sense like that..